A1-http-connect-proxy-support.md

```diff
@@ -1,222 +1,242 @@
-proxy instead of a TCP-level proxy, where the proxy natively speaks gRPC.
-The idea is that the proxy accepts a gRPC request from the external client
-and then uses its own client on the internal side, which does the name
-resolution and per-call load balancing internally to the backend servers.
-The trade-off here is that this approach requires that the gRPC-level
-proxy be trusted by all internal servers, which has security implications
-(e.g., a successful attack on the proxy would yield priviledged access
-on all internal servers).
+mapper and the name resolver both have to agree on the list of known proxy
+addresses. So, for example, if the proxy mapper implementation is getting the
+list of proxy addresses from a local file, the service owner would need to first
+push an updated list that contains the new proxy address to all clients. Then,
+once all clients have been updated, the new proxy can be added to DNS.
+
+Note that another alternative for case 3 would be to use a gRPC-level proxy
+instead of a TCP-level proxy, where the proxy natively speaks gRPC. The idea is
+that the proxy accepts a gRPC request from the external client and then uses its
+own client on the internal side, which does the name resolution and per-call
+load balancing internally to the backend servers. The trade-off here is that
+this approach requires that the gRPC-level proxy be trusted by all internal
+servers, which has security implications (e.g., a successful attack on the proxy
+would yield priviledged access on all internal servers).
 
 ## Implementation
 
-In C-core, the HTTP CONNECT handshaker was already implemented.
-The following additional changes have also been made:
+### C-Core
+
+In C-core, the HTTP CONNECT handshaker was already implemented. The following
+additional changes have also been made:
 
-- [grpc/grpc#9383](https://github.com/grpc/grpc/pull/9383) (changes
-  HTTP CONNECT handshaker to be triggered via channel args)
-- [grpc/grpc#9372](https://github.com/grpc/grpc/pull/9372) (adds proxy
-  mapper hook for address rewriting)
-- [grpc/grpc#9557](https://github.com/grpc/grpc/pull/9557) (adds proxy
-  mapper hook for name rewriting)
+-   [grpc/grpc#9383](https://github.com/grpc/grpc/pull/9383) (changes HTTP
+    CONNECT handshaker to be triggered via channel args)
+-   [grpc/grpc#9372](https://github.com/grpc/grpc/pull/9372) (adds proxy mapper
+    hook for address rewriting)
+-   [grpc/grpc#9557](https://github.com/grpc/grpc/pull/9557) (adds proxy mapper
+    hook for name rewriting)
+
+C-Core checks the following places to determine the HTTP proxy to use, stopping
+at the first one that is set:
+
+1.  `GRPC_ARG_HTTP_PROXY` channel arg
+2.  `grpc_proxy environment` variable
+3.  `https_proxy` environment variable
+4.  `http_proxy` environment variable
+
+If none of the above are set, then no HTTP proxy will be used.
+
+If an HTTP proxy to be used is found, C-Core then checks the follows places to
```

Comment:
s/the follows places/the following places/

Reply:
In "If an HTTP proxy to be used is found" what is "found"? Or are you just saying "if an HTTP proxy is to be used as dictated by the channel args or environment. variables"?

Reply:
Updated